src/services/safeBlock.ts

```diff
@@ -0,0 +1,34 @@
+import config from "config";
+
+import { Pool } from "pg";
+import { Request, Response } from "express";
+
+import { CardanoFrag } from "../Transactions/types";
+
+const safeBlockQuery = `SELECT epoch_no AS "epoch",
+    epoch_slot_no AS "slot",
+    encode(hash, 'hex') as hash,
+    block_no AS height
+FROM BLOCK
+WHERE block_no <= (SELECT MAX(block_no) FROM block) - ($1)::int
+ORDER BY id DESC
+LIMIT 1;
+`;
+
+export const handleSafeBlock = (pool: Pool) => async (req: Request, res: Response) => {
+    const safeBlockDifference = parseInt(config.get("safeBlockDifference"));
+
+    const result = await pool.query(safeBlockQuery, [safeBlockDifference]);
+    if (result.rowCount === 0) throw new Error("no blocks found!");
+
+    const row = result.rows[0];
+
+    const safeBlock: CardanoFrag = {
+        epoch: row.epoch,
+        slot: row.slot,
+        hash: row.hash,
+        height: row.height
+    };
+
+    res.send(safeBlock);
```

Comment:
Let's make the return value to be exactly compatible with the `/bestblock` endpoint, so it would be like: `{ kind: 'ok', value: safeBlock }`

Will be more considetent with the existing API and also a better option for potential extending

Reply:
The response is actually exactly the same as from `bestblock`. I think there was a confusion because the function `askBestBlock` at `bestblock.ts` is just an intermediary function that is used in the actual request handler. If you look at the `src/index.ts` file, you will see the actual handler of the `/bestblock` request is the `bestBlock` function, which internally uses `askBestBlock` and then based on the `kind` field returned by `askBestBlock`, it either just sends back the actual block information in case `kind` is `ok`, or throw an error otherwise.

Reply:
Yup, that's a good point, I missed thi, let's not change that then